### PR TITLE
More base conversions

### PR DESCRIPTION
**Pull request by yminsky**
Move the first-class-modules and functors chapters to Base, mostly.

### CODE REVIEW

File: examples/code/fcm/query_handler.mlt [hunk 1]
```diff
@@ -52,7 +52,7 @@ val sexp_of_u : u -> Sexp.t = <fun>
 |}];;
 sexp_of_u {a=3;b=7.};;
 [%%expect ocaml {|- : Sexp.t = ((a 3) (b 7))|}];;
-u_of_sexp (Sexp.of_string "((a 43) (b 3.4))");;
+u_of_sexp (Core_kernel.Sexp.of_string "((a 43) (b 3.4))");;
```
Comment (avsm) [inline examples/code/fcm/query_handler.mlt]: Why is Core_kernel still needed here?  I thought Sexp was part of Base...

File: examples/code/fcm/query_handler.mlt [hunk 2]
```diff
@@ -206,13 +206,16 @@ val dispatch :
   Sexp.t -> Sexp.t Or_error.t = <fun>
 |}];;
 [@@@part "13"];;
+open Stdio;;
 let rec cli dispatch_table =
   printf ">>> %!";
   let result =
-    match In_channel.input_line stdin with
+    match In_channel.input_line In_channel.stdin with
```
Comment (avsm) [inline examples/code/fcm/query_handler.mlt]: `match In_channel.(input_line stdin)` instead?

File: examples/code/fcm/query_handler.mlt (hunk 1 above)
Comment (yminsky) [reply examples/code/fcm/query_handler.mlt]: Base has s-expressions, but no printers or parsers. Those are in Parsexp and Sexplib. Core_kernel does include those. Maybe we should reference Parsexp directly, but the API is a little more complicated.

File: examples/code/fcm/query_handler.mlt (hunk 2 above)
Comment (yminsky) [reply examples/code/fcm/query_handler.mlt]: A good idea. I'll do that.

**Pull request merged**